server/core/src/https/mod.rs

```diff
@@ -460,65 +458,53 @@ pub(crate) async fn handle_tls_conn(
     let (stream, client_addr) =
         process_client_addr(stream, connection_addr, trusted_proxy_v2_ips).await?;
 
-    let ssl = Ssl::new(acceptor.context()).map_err(|e| {
-        error!("Failed to create TLS context: {:?}", e);
-        std::io::Error::from(ErrorKind::ConnectionAborted)
-    })?;
-
-    let mut tls_stream = SslStream::new(ssl, stream).map_err(|err| {
+    let tls_stream = acceptor.accept(stream).await.map_err(|err| {
         error!(?err, "Failed to create TLS stream");
         std::io::Error::from(ErrorKind::ConnectionAborted)
     })?;
 
-    match SslStream::accept(Pin::new(&mut tls_stream)).await {
-        Ok(_) => {
-            // Process the client cert (if any)
-            let client_cert = if let Some(peer_cert) = tls_stream.ssl().peer_certificate() {
-                // TODO: This is where we should be checking the CRL!!!
-
-                // Extract the cert from openssl to x509-cert which is a better
-                // parser to handle the various extensions.
-
-                let cert_der = peer_cert.to_der().map_err(|ossl_err| {
-                    error!(?ossl_err, "unable to process x509 certificate as DER");
-                    std::io::Error::from(ErrorKind::ConnectionAborted)
-                })?;
-
-                let certificate = Certificate::from_der(&cert_der).map_err(|ossl_err| {
-                    error!(?ossl_err, "unable to process DER certificate to x509");
-                    std::io::Error::from(ErrorKind::ConnectionAborted)
-                })?;
-
-                let public_key_s256 = x509_public_key_s256(&certificate).ok_or_else(|| {
-                    error!("subject public key bitstring is not octet aligned");
-                    std::io::Error::from(ErrorKind::ConnectionAborted)
-                })?;
-
-                Some(ClientCertInfo {
-                    public_key_s256,
-                    certificate,
-                })
-            } else {
-                None
-            };
-
-            let client_conn_info = ClientConnInfo {
-                connection_addr,
-                client_addr,
-                client_cert,
-            };
-
-            // Hyper has its own `AsyncRead` and `AsyncWrite` traits and doesn't use tokio.
-            // `TokioIo` converts between them.
-            let stream = TokioIo::new(tls_stream);
-
-            process_client_hyper(stream, app, client_conn_info).await
-        }
-        Err(error) => {
-            trace!("Failed to handle connection: {:?}", error);
-            Ok(())
-        }
-    }
+    let maybe_peer_cert = tls_stream
+        .get_ref()
+        .1
+        .peer_certificates()
+        // The first certificate relates to the peer.
+        .and_then(|peer_certs| peer_certs.first());
+
+    // Process the client cert (if any)
+    let client_cert = if let Some(peer_cert) = maybe_peer_cert {
+        // TODO: This is where we should be checking the CRL!!!
```

Comment:
Can you please make this an explicit GitHub issue referenced from this line, since that's a security-level TODO, and these things get embarrassing later 😄